### PR TITLE
Include optional navigation tree in JSON token schema to support custom navigation panel

### PR DESCRIPTION
**Pull request by praveenkuttappan**
APIView Token schema changes to support custom navigation panel

### CODE REVIEW

File: tools/apiview/parsers/apiview-treestyle-parser-schema/main.tsp [hunk 1]
```diff
@@ -18,6 +18,10 @@ model CodeFile {
   ReviewLines: Array<ReviewLine>;
   /** Add any system generated comments. Each comment is linked to review line ID */
   Diagnostics?: Array<CodeDiagnostic>;
+  /** Navigation items are used to create a tree view in the navigation panel. Each navigation item is linked to a review line ID. this is optional.
```
Comment (maririos) [inline tools/apiview/parsers/apiview-treestyle-parser-schema/main.tsp:21]:
```suggestion
  /** Navigation items are used to create a tree view in the navigation panel. Each navigation item is linked to a review line ID. This is optional.
```

**Comment by maririos** [comment]
Didn't you add a pipeline that checked that the swagger was up to date with the tsp?

File: tools/apiview/parsers/apiview-treestyle-parser-schema/main.tsp [hunk 2]
```diff
@@ -100,4 +104,11 @@ enum CodeDiagnosticLevel {
   /** Fatal level diagnostic will block API review approval and it will show an error message to the user. Approver will have to 
   * override fatal level system comments before approving a review.*/
  Fatal:	4
-}
\ No newline at end of file
+}
+
+model NavigationItem {
```
Comment (chidozieononiwu) [inline tools/apiview/parsers/apiview-treestyle-parser-schema/main.tsp:109]: Will you also support custom icons?

Comment (praveenkuttappan) [reply tools/apiview/parsers/apiview-treestyle-parser-schema/main.tsp:109]: That can already be done Tags in this model.

**Comment by praveenkuttappan** [comment]
> Didn't you add a pipeline that checked that the swagger was up to date with the tsp?

https://dev.azure.com/azure-sdk/public/_build?definitionId=7166

File: tools/apiview/parsers/apiview-treestyle-parser-schema/CodeFile.json [hunk 3]
```diff
@@ -98,6 +98,13 @@
                         "$ref": "CodeDiagnostic.json"
                     },
                     "description": "Add any system generated comments. Each comment is linked to review line ID"
+                },
+                "Navigation": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "NavigationItem.json"
+                    },
+                    "description": "Navigation items are used to create a tree view in the navigation panel. Each navigation item is linked to a review line ID. this is optional.\nIf navigation items are not provided then navigation panel will be automatically generated using the review lines. Navigation items should be provided only if you want to customize the navigation panel."
```
Comment (praveenkuttappan) [inline tools/apiview/parsers/apiview-treestyle-parser-schema/CodeFile.json:107]:
```suggestion
                    "description": "Navigation items are used to create a tree view in the navigation panel. Each navigation item is linked to a review line ID. This is optional.\nIf navigation items are not provided then navigation panel will be automatically generated using the review lines. Navigation items should be provided only if you want to customize the navigation panel."
```

**Review by maririos — APPROVED** [review]

**Pull request merged**